av_metrics_tool/src/main.rs

```diff
@@ -247,6 +270,28 @@ fn run_video_metrics(
     results
 }
 
+#[inline(always)]
+fn count_frames<D: Decoder, P: Pixel>(dec1: &mut D, dec2: &mut D) -> u64 {
+    let mut frame_number = 0;
+
+    while dec1.read_video_frame::<P>().is_some() && dec2.read_video_frame::<P>().is_some() {
```

Comment:
I think in IRC it was mentioned there is a way you could estimate the size without reading through the entire file.

(Size of file in bytes - size of y4m headers) / (size of one frame in bytes)

Let me know if this works,, the only situation where it might not is if y4m supports changing resolution or bit-depth mid encode but I don't think that it does.

Reply:
Yep, I could try that, but what would happen if someone decides to use the `ffmpeg` decoder? How should we treat that situation?

Reply:
Ah, that's a good point... In that case we could make a more optimized implementation to get used if we're reading a y4m, or I'm also just fine with keeping this for now.

Reply:
I agree, we could iterate over this implementation and differentiate that according to the decoder, thanks a lot for the review! :)